packages/babel-plugin-transform-react-constant-elements/src/index.js

```diff
@@ -101,7 +102,11 @@ export default function transformReactConstantElement(api, options) {
         path.traverse(immutabilityVisitor, state);
 
         if (state.isImmutable) {
-          path.hoist();
+          const hoisted = path.hoist();
+          // instanceof NodePath
+          if (hoisted instanceof path.constructor) {
```

Comment:
Why not `NodePath`? Is the `instanceof` even necessary?

Reply:
just lazy before hitting the bed and I didn't search from where I could import this

`path.hoist()` can return undefined, that is a guard against that - I guess I could reverse the condition and remove 'dep' on the `instanceof` altogether

Reply:
If it's only `undefined` or a path, why not just truthy check `hoisted`?

```js
if (hoisted) {
  //...
}

Reply:
Fixed